engine/crates/federated-graph/src/federated_graph/v3.rs

```diff
@@ -208,6 +208,81 @@ impl From<super::v2::FederatedGraphV2> for FederatedGraphV3 {
             }
         }
 
+        // In FederatedGraphV3, we reserve two fields for __schema and __type
+        // on the root Query type. The space for them needs to be created. This
+        // block is responsible for it.
+        {
+            let new_fields = ["__schema", "__type"].map(|needle| {
+                value
+                    .strings
+                    .iter()
+                    .position(|string| string == needle)
+                    .map(StringId)
+                    .unwrap_or_else(|| {
+                        let idx = value.strings.len();
+                        value.strings.push((*needle).to_owned());
+                        StringId(idx)
+                    })
+            });
+
+            let query_object_id = value.root_operation_types.query;
+            let original_start = {
+                let query_object_fields = object_fields[query_object_id.0].as_mut().expect("Query to have fields");
```

Comment:
Is this expect here because the query must have fields at this point?

Reply:
yes, we added two so we are expecting at least these.